tornjak-frontend/src/components/agent-list.tsx

```diff
@@ -58,34 +75,31 @@ class AgentList extends Component {
       this.TornjakApi.refreshLocalSelectorsState(this.props.agentworkloadSelectorInfoFunc);
       this.TornjakApi.populateLocalAgentsUpdate(this.props.agentsListUpdateFunc, this.props.tornjakMessageFunc);
       this.TornjakApi.populateLocalTornjakServerInfo(this.props.tornjakServerInfoUpdateFunc, this.props.tornjakMessageFunc);
-      if(this.props.globalTornjakServerInfo !== "") {
+      if (this.props.globalTornjakServerInfo && Object.keys(this.props.globalTornjakServerInfo).length) {
```

Comment:
@lumjjb Brandon the code above is: 
`this.props.globalTornjakServerInfo` this check for undefined, null, empty string, 0 and false
`Object.keys(this.props.globalTornjakServerInfo).length` this checks if the object has any properties.
Objects and types derived from object can not be checked if assigned by this method as they are pointers and returns the address of the object or derived object. In case of objects we check for properties, in case of arrays we check for length and etc.

Reply:
Hmm I guess my question would be when would this be the case? And if so, should we do this for every undefined check? Maybe make a helper to do check of undefined and its properties?

Reply:
@florentinvintila is right, this would check if the object is defined/ initialized. We can use it for the case of undefined as you suggested depending on the return type @lumjjb. I will go through this case by case and check how we can generalize.